static/app/components/profiling/flamegraph/flamegraphSpans.tsx

```diff
@@ -20,14 +33,25 @@ interface FlamegraphSpansProps {
 export function FlamegraphSpans({
   spanChart,
   canvasPoolManager,
+  canvasBounds,
   spansView,
   spansCanvas,
   spansCanvasRef,
   setSpansCanvasRef,
 }: FlamegraphSpansProps) {
+  const dispatch = useDispatchFlamegraphState();
   const flamegraphTheme = useFlamegraphTheme();
   const scheduler = useMemo(() => new CanvasScheduler(), []);
 
+  const [configSpaceCursor, setConfigSpaceCursor] = useState<vec2 | null>(null);
+  const [startPanVector, setStartPanVector] = useState<vec2 | null>(null);
+  const [lastInteraction, setLastInteraction] = useState<
+    'pan' | 'click' | 'zoom' | 'scroll' | 'select' | 'resize' | null
+  >(null);
+
+  const previousInteraction = usePrevious(lastInteraction);
+  const beforeInteractionConfigView = useRef<Rect | null>(null);
+
```

Comment:
All of the lines here were copy pasted from flamegraphzoomView... Once I'm done with this feature I will abstract them into their own hooks so we can cleanup the components